src/input/type_base.cc

```diff
@@ -190,6 +194,40 @@ bool Array::valid_default(const string &str) const {
 }
 
 
+boost::shared_ptr<TypeBase> Array::make_instance(std::vector<ParameterPair> vec) const {
+	// Replace only if type_of_values_ is parameter
+	if ( typeid( *(data_->type_of_values_) ) == typeid(Parameter) ) {
+		ASSERT( vec.size() == 1, "Parameter vector of Input::Type::Array '%s' must have size 1!", this->type_name().c_str());
```

Comment:
vec.size()>1 is valid; need parameter name lookup same as in Record::make_instance ...
rather implement Parameter::make_instance which makes parameter name lookup.

Carefully think what we have to do with the type substituted for the parameter. Surly call finish, maybe call make_instance?